e2e-tests/tests/test_analyzer.py

```diff
@@ -480,3 +480,94 @@ def test_given_ad_hoc_deny_list_recognizer_the_right_entities_are_returned():
     assert equal_json_strings(
         expected_response, response_content, ignore_keys=["recognition_metadata"]
     )
+
+
+@pytest.mark.api
+def test_given_allow_list_then_no_entity_is_returned():
+    request_body = """
+    {
+        "text": "email: admin@github.com", 
+        "language": "en", 
+        "allow_list": ["admin@github.com"]
+    }
+    """
+
+    response_status, response_content = analyze(request_body)
+
+    expected_response = """
+     []
+    """
+    assert response_status == 200
+    assert equal_json_strings(
+        expected_response, response_content
+    )
+
+
+@pytest.mark.api
+def test_given_allow_list_with_regex_match_then_no_entity_is_returned():
+    request_body = """
+    {
+        "text": "email: admin@github.com", 
+        "language": "en", 
+        "allow_list": [".*@github.com"],
+        "allow_list_match": "regex"
+    }
+    """
+
+    response_status, response_content = analyze(request_body)
+
+    expected_response = """
+     []
+    """
+    assert response_status == 200
+    assert equal_json_strings(
+        expected_response, response_content
+    )
+
+
+@pytest.mark.api
+def test_given_allow_list_without_setting_allow_list_match_then_normal_entity_is_returned():
+    request_body = """
+    {
+        "text": "email: admin@github.com", 
+        "language": "en", 
+        "allow_list": [".*@github.com"]
+    }
+    """
+
+    response_status, response_content = analyze(request_body)
+
+    expected_response = """
+     [
+         {"entity_type": "EMAIL_ADDRESS", "start": 7, "end": 23, "score": 0.85, "analysis_explanation":null}
+     ]
+    """
+    assert response_status == 200
+    assert equal_json_strings(
+        expected_response, response_content, ignore_keys=["recognition_metadata"]
+    )
+
+
+@pytest.mark.api
+def test_given_regex_flags_and_normal_entities_are_returned():
+    request_body = """
+    {
+        "text": "email: admin@GitHub.com", 
+        "language": "en", 
+        "allow_list": [".*@github.com"],
+        "allow_list_match": "regex",
+        "regex_flags": 0
+    }
+    """
+
+    response_status, response_content = analyze(request_body)
+
+    expected_response = """
+     [
```

Comment:
Why is the email returned in this case?

Reply:
because the regex_flags is set to be case sensitive in this case and it will not work in this case, let me elaborate by following code:
```
In [1]: import re

In [2]: re_compiled = re.compile(".*github.com", flags=(re.DOTALL | re.MULTILINE | re.IGNORECASE))

In [3]: text = "email: admin@GitHub.com"

In [4]: re_compiled.search(text)
Out[4]: <re.Match object; span=(0, 23), match='email: admin@GitHub.com'>

In [5]: re_compiled = re.compile(".*github.com", flags=0)

In [6]: re_compiled.search(text)
```

Reply:
I see. Thanks. Could you please add a comment? It's a bit hard to spot.